Doc/using/cmdline.rst

```diff
@@ -994,3 +994,71 @@ Debug-mode variables
    Need Python configured with the :option:`--with-trace-refs` build option.
 
    .. versionadded:: 3.11
+
+.. _using-on-finding-modules:
+
+Finding modules
+---------------
+
+A module search path is initialized when Python starts. This module search path
+may be accessed at :data:`sys.path`.
+
+The first entry in the module search path is the directory containing the input
+script (or the the current directory when no file is specified).
+
+On Windows a default Python installation adds the :file:`/Lib` directory and
+the :file:`/Lib/site-packages` directory to the module search path. These
+locations are relative to the :file:`python.exe` location. Standard modules
+are usually found in the Lib directory and third-party modules in the
+Lib/site-packages directory.
+
+On Unix the lib and lib/site-packages directories are added by looking for
+:file:`/exec_prefix/lib/pythonversion` , :file:`/exec_prefix/lib/pythonversion/site-packages` ,
+:file:`/prefix/lib/pythonversion` and :file:`/prefix/lib/pythonversion/site-packages` .
+Often ``exec_prefix`` and ``prefix`` are the same and a common result is
+adding :file:`/usr/lib/pythonversion` and :file:`/usr/lib/pythonversion/site-packages`
+to the module search path. See :ref:`using-on-unix` for details.
+
+The :envvar:`PYTHONPATH` environment variable is often used to add directories
+to the search path.
+
+Another common way to set the search path is to create :mod:`sitecustomize`
+or :mod:`usercustomize` modules as described in the :mod:`site` module
+documentation.
+
+Virtual environments
+~~~~~~~~~~~~~~~~~~~~
+
+If Python is run in a virtual environment (as described at :ref:`tut-venv`)
+the Lib and Lib/site-packages directories are specific to the virtual
+environment.
+
+If a ``pyvenv.cfg`` file is found alongside the main executable, or in the
+directory one level above the executable, the following variations apply:
+
+* If ``home`` is an absolute path and :envvar:`PYTHONHOME` is not set, this
+  path is used instead of the path to the main executable when deducing the
+  Lib and Lib/site-packages directory locations.
+
+_pth files
+~~~~~~~~~~
+
+To completely override :data:`sys.path`, create a ``._pth`` file with the same
+name as the DLL/dylib/so (``python311._pth``) or the executable (``python._pth``)
```

Comment:
> same name as the DLL/dylib/so

Currently only the DLL/dylib work - for POSIX we don't have an (obvious) way to get the real `.so`, and generally there isn't one (see `library_to_dict` in getpath.c). We could probably get the name of it at compile time, but so many builds deliberately move thing around afterwards that it wouldn't be trustworthy for this feature.

Reply:
Thank you, I know Windows very well but my POSIX knowledge is dismal.

Reply:
It's not obvious without walking through the source code anyway. Hence, we need this doc :)